app/main.py

```diff
@@ -1,6 +1,41 @@
 from typing import Callable
+import functools
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_storage = {}  # Dictionary to store results of completed runs
```

Comment:
The cache_storage dictionary is mutable and is used in the inner function, so there is no need for the nonlocal statement. Python allows you to modify mutable objects in the inner function without declaring them as nonlocal.